pkg/cmd/pulumi/packagecmd/package_publish.go

```diff
@@ -236,3 +247,57 @@ func login(ctx context.Context, project *workspace.Project) (backend.Backend, er
 
 	return b, nil
 }
+
+// findReadme attempts to find a readme file in the given package source.
+// It tries to find a readme in the following order and returns the path to the first one it finds:
+// 1. The package source if it is a directory
+// 2. The installed plugin directory
+// If no readme is found, an empty string is returned.
```

Comment:
I think we should call out that it's specifically a `readme.md` we're after, and that the search is not case sensitive.